File: 1716. Calculate Money in Leetcode Bank/Solution.cpp
// Author        : tarlansoltanov
// Title         : Calculate Money in Leetcode Bank
// ID            : 1716
// URL           : https://leetcode.com/problems/calculate-money-in-leetcode-bank/
// Difficulty    : Easy
// Language      : C++


class Solution {
public:
    int totalMoney(int n) {
        int k = 0, t = 0, res = 0;
        while(n--){
          res += ++k;
          t = k;
          for(int i = 0; i < 6; i++){
            if(n == 0) return res;
            res += ++t;
            n--;
          }
        }
        return res;
    }
};

File: 1716. Calculate Money in Leetcode Bank/Solution.py
# Author        : tarlansoltanov
# Title         : Calculate Money in Leetcode Bank
# ID            : 1716
# URL           : https://leetcode.com/problems/calculate-money-in-leetcode-bank/
# Difficulty    : Easy
# Language      : Python


class Solution:
  def totalMoney(self, n: int) -> int:
    k, res = 0, 0

    while n > 0:
      n, k, res = n - 1, k + 1, res + k + 1

      for i in range(6):
        if n == 0: return res
        res += k + i + 1
        n -= 1
      

    return res